pennylane/ops/op_math/decompositions/controlled_decompositions.py

```diff
@@ -217,6 +220,21 @@ def _ctrl_decomp_bisect_resources(num_target_wires, num_control_wires, **__):
     len_k1 = (num_control_wires + 1) // 2
     len_k2 = num_control_wires - len_k1
     # this is a general overestimate based on the resource requirement of the general case.
+    if len_k1 == len_k2:
+        return {
+            resource_rep(ops.QubitUnitary, num_wires=num_target_wires): 4,
+            adjoint_resource_rep(ops.QubitUnitary, {"num_wires": num_target_wires}): 4,
+            controlled_resource_rep(
+                ops.X,
+                {},
+                num_control_wires=len_k2,
+                num_work_wires=len_k1,
+                work_wire_type="dirty",
```

Comment:
Could you also explain what "dirty" does here?

Reply:
See my comment in the other thread